src/pages/ReimbursementAccount/ReimbursementAccountPage.tsx

```diff
@@ -205,6 +195,16 @@ function ReimbursementAccountPage({route, policy}: ReimbursementAccountPageProps
         return achData?.state === BankAccount.STATE.PENDING || [CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT, ''].includes(getStepToOpenFromRouteParams(route));
     }
 
+    /**
+     When this page is first opened, `reimbursementAccount` prop might not yet be fully loaded from Onyx.
+     Calculating `shouldShowContinueSetupButton` immediately on initial render doesn't make sense as
+     it relies on incomplete data. Thus, we should wait to calculate it until we have received
+     the full `reimbursementAccount` data from the server. This logic is handled within the useEffect hook,
+     which acts similarly to `componentDidUpdate` when the `reimbursementAccount` dependency changes.
+     */
+    const [hasACHDataBeenLoaded, setHasACHDataBeenLoaded] = useState(reimbursementAccount !== CONST.REIMBURSEMENT_ACCOUNT.DEFAULT_DATA && isPreviousPolicy);
+    const [shouldShowContinueSetupButton, setShouldShowContinueSetupButton] = useState(getShouldShowContinueSetupButtonInitialValue);
```

Comment:
No I think? I'm able to connect via Plaid

Reply:
One suggestion, you also apply the same fix as we did [here](https://github.com/Expensify/App/pull/51315) for this bug only

Reply:
Yeah, just moving the declaration above, before these functions are called.

Changing to `!requestorStepRef || !requestorStepRef.current` gives a lot of lint errors. Do you think I should still do it and suppress them? In case these functions will be moved back in the future

Reply:
> One suggestion, you also apply the same fix as we did [here](https://github.com/Expensify/App/pull/51315) for this bug only

@DylanDylann I misunderstood you on this, do you want me to also do useEffect and useCallback changes from that PR? It wasn't the reason it got reverted?